local.go

```diff
@@ -246,60 +272,64 @@ func getStartingLocation(p *Problem, method Method, initX []float64, stats *Stat
 			loc.Hessian.CopySym(initH)
 		}
 	} else {
-		evalType = FuncEvaluation
+		eval := FuncEvaluation
 		if loc.Gradient != nil {
-			evalType |= GradEvaluation
+			eval |= GradEvaluation
 		}
 		if loc.Hessian != nil {
-			evalType |= HessEvaluation
+			eval |= HessEvaluation
 		}
-		evaluate(p, evalType, loc.X, loc, stats)
+		evaluate(p, loc, eval, stats)
 	}
 
 	if math.IsNaN(loc.F) {
-		return loc, evalType, ErrNaN
+		return loc, ErrNaN
 	}
 	if math.IsInf(loc.F, 1) {
-		return loc, evalType, ErrInf
+		return loc, ErrInf
 	}
 	for _, v := range loc.Gradient {
 		if math.IsInf(v, 0) {
-			return loc, evalType, ErrGradInf
+			return loc, ErrGradInf
 		}
 		if math.IsNaN(v) {
-			return loc, evalType, ErrGradNaN
+			return loc, ErrGradNaN
 		}
 	}
 
-	return loc, evalType, nil
+	return loc, nil
 }
 
-func checkConvergence(loc *Location, iterType IterationType, stats *Stats, settings *Settings) Status {
-	if iterType == MajorIteration || iterType == InitIteration {
-		if loc.Gradient != nil {
-			norm := floats.Norm(loc.Gradient, math.Inf(1))
-			if norm < settings.GradientThreshold {
-				return GradientThreshold
-			}
-		}
-		if loc.F < settings.FunctionThreshold {
-			return FunctionThreshold
+func checkConvergence(loc *Location, settings *Settings) Status {
+	if loc.Gradient != nil {
+		norm := floats.Norm(loc.Gradient, math.Inf(1))
+		if norm < settings.GradientThreshold {
+			return GradientThreshold
 		}
 	}
 
-	if iterType == MajorIteration && settings.FunctionConverge != nil {
-		status := settings.FunctionConverge.FunctionConverged(loc.F)
-		if status != NotTerminated {
-			return status
-		}
+	if loc.F < settings.FunctionThreshold {
+		return FunctionThreshold
+	}
+
+	if settings.FunctionConverge != nil {
+		return settings.FunctionConverge.FunctionConverged(loc.F)
 	}
 
+	return NotTerminated
+}
+
+func checkLimits(loc *Location, stats *Stats, settings *Settings) Status {
```

Comment:
Please add a comment that this distinguishing feature between checkLimits and checkConvergence is that checkLimits is checked every evaluation, while checkConvergence is only examined at MajorIteration.